internal/transport/transport.go

```diff
@@ -514,6 +501,30 @@ func Truncate(ctx context.Context, cfg *Config) error {
 	return nil
 }
 
+// Truncate will truncate the defined tables in the configuration.
+func Truncate(ctx context.Context, cfg *Config) error {
+	// truncateRequest is a special request that will truncate the table before upserting data.
+	truncateRequest := new(proto.TruncateRequest)
+
+	if cfg.Truncate {
+		for _, req := range cfg.Requests {
+			// Add the table to the list of tables to truncate.
+			if req.Truncate {
```

Comment:
Ah so this is a little tricky 🤔 We have to do one more thing. We should change the type for `Request.Truncate` to `*bool` from `bool` and here do this:

```go
if req.Truncate == nil && *req.Truncate { }
```

And on L519 do this:
```go
if table := req.Table; req.Truncate != nil && *req.Truncate && table != "" {}
```

Reply:
@s4kh The table should never be blank because of [this logic](https://github.com/alpine-hodler/gidari/blob/main/internal/transport/transport.go#L213).

But a bool defaults too false, so even if the user doesn't set one it will be false. Which makes this check kind of tedious.